.github/workflows/src/summarize-checks.js

```diff
@@ -0,0 +1,856 @@
+ * @returns {Promise<[string[], string[]]>}
+ */
+export async function handleLabeledEvent(
+  { context, core },
+  owner,
+  repo,
+  issue_number,
+  event_name,
+  known_labels,
+) {
+  // logic for this event is based on code directly ripped from pipelinebot:
+  // private/openapi-kebab/src/bots/pipeline/pipelineBotOnPRLabelEvent.ts
+  // todo: further enhance with labelling actions from `PR Summary` check.
+  const changedLabel = context.payload.label?.name;
+  const labelsToAdd = new Set();
+  const labelsToRemove = new Set();
+
+  if (event_name === "labeled") {
+    if (changedLabel == "ARMChangesRequested") {
+      if (known_labels.indexOf("WaitForARMFeedback") !== -1) {
+        labelsToRemove.add("WaitForARMFeedback");
+      }
+    }
+    if (changedLabel == "ARMSignedOff") {
+      if (known_labels.indexOf("WaitForARMFeedback") !== -1) {
+        labelsToRemove.add("WaitForARMFeedback");
+      }
+      if (known_labels.indexOf("ARMChangesRequested") !== -1) {
+        labelsToRemove.add("ARMChangesRequested");
+      }
+    }
+
+    for (const label of labelsToRemove) {
+      core.info(`Removing label: ${label} from ${owner}/${repo}#${issue_number}.`);
+      // await github.rest.issues.removeLabel({
+      //   owner: owner,
+      //   repo: repo,
+      //   issue_number: issue_number,
+      //   name: label,
+      // });
+    }
+  } else if (event_name === "unlabeled") {
+    if (changedLabel == "ARMChangesRequested") {
+      if (known_labels.indexOf("WaitForARMFeedback") !== -1) {
+        labelsToAdd.add("WaitForARMFeedback");
+      }
+    }
+
+    if (labelsToAdd.size > 0) {
+      core.info(
+        `Adding labels: ${Array.from(labelsToAdd).join(", ")} to ${owner}/${repo}#${issue_number}.`,
+      );
+      // await github.rest.issues.addLabels({
+      //   owner: owner,
+      //   repo: repo,
+      //   issue_number: issue_number,
+      //   labels: Array.from(labelsToAdd),
+      // });
+    }
+  }
+
+  return [Array.from(labelsToAdd), Array.from(labelsToRemove)];
+}
+
+// #endregion
+// #region checks
+/**
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ * @param {string} owner - The repository owner.
+ * @param {string} repo - The repository name.
+ * @param {string} head_sha - The commit SHA to check.
+ * @param {number} prNumber - The pull request number.
+ * @param {string[]} excludedCheckNames
+ * @returns {Promise<[CheckRunData[], CheckRunData[]]>}
+ */
+export async function getCheckRunTuple(
+  { github, core },
+  owner,
+  repo,
+  head_sha,
+  prNumber,
+  excludedCheckNames,
+) {
+  // This function was originally a version of getRequiredAndFyiAndAutomatedMergingRequirementsMetCheckRuns
+  // but has been simplified for clarity and purpose.
+  /** @type {CheckRunData[]} */
+  let reqCheckRuns = [];
+  /** @type {CheckRunData[]} */
+  let fyiCheckRuns = [];
+
+  const response = await github.graphql(getGraphQLQuery(owner, repo, head_sha, prNumber));
```

Comment:
Why GraphQL instead of REST?

Reply:
There is no way to get checksuites from REST.

Reply:
https://docs.github.com/en/rest/checks?apiVersion=2022-11-28

We use "List check runs for a Git reference" (below), so I suspect "List check suites for a Git reference" would also work.

I'm not opposed to GraphQL, other than all our existing code in this repo uses REST.  One the one hand, it might be good to get our feet wet with GraphQL.  On the other hand, I would prefer to use one or the other as consistently as possible in our code for ease of maintenance.

You can merge this PR with GraphQL, but I might submit a PR converting it to REST, if I think it looks better.

https://github.com/Azure/azure-rest-api-specs/blob/89a7e31c277506592a7d4d273b852ca5a5cc324c/.github/workflows/src/github.js#L156